attestation-agent/attester/src/lib.rs

```diff
@@ -211,5 +211,13 @@ pub fn detect_attestable_devices() -> Vec<Tee> {
         additional_devices.push(Tee::HygonDcu);
     }
 
+    #[cfg(feature = "tpm-attester")]
+    if detect_tee_type() != Tee::Tpm {
```

Comment:
You may want to add a comment here. It looks like this will trigger only if TPM is enabled as well as some other cpu platform (besides the sample one) and it depends on the TPM check being last in `tee_type_check` which could change in the future. Maybe add a comment there too about how it should be last.

Keep in mind that there is currently no binding between the TPM and the CPU platform so whatever your working on here probably is not secure. I guess that is potentially ok, but there is some risk to adding features that people might not realize are experimental.